Backpack/TappableLinkLabel/Classes/BPKTappableLinkLabel.m

```diff
@@ -167,16 +167,33 @@ - (void)updateTextDisplay {
         self.contentView.text = nil;
         return;
     }
-
-    NSDictionary<NSAttributedStringKey, id> *newStringAttributes = [BPKFont attributesForFontStyle:self.fontStyle
-                                                                              withCustomAttributes:self.customFontAttributes];
-
-    NSAttributedString *newString = [[NSAttributedString alloc] initWithString:self.text attributes:newStringAttributes];
-    self.contentView.text = newString;
-
+    
+    
+//    NSMutableAttributedString *attText = [[BPKFont attributedStringWithFontStyle:self.fontStyle content:self.text textColor:textColor] mutableCopy];
+//    self.contentView.kern = 0;
+    
+
+    UIFont *font = [BPKFont fontForFontStyle:self.fontStyle];
+    
     // Note: we have to set the font on our TTTAttributedLabel as this is what will be used in calculating its size.
-    self.contentView.font = newStringAttributes[NSFontAttributeName];
-
+    self.contentView.font = font;
+    self.contentView.kern = [[BPKFont letterSpacingForStyle:self.fontStyle] doubleValue];
+    
+    NSMutableParagraphStyle *paragraphStyle = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
+    
+    CGFloat lineHeight = [BPKFont lineHeightForStyle:self.fontStyle];
+    [paragraphStyle setLineSpacing:lineHeight - font.lineHeight];
+    
+    NSMutableDictionary *attributes = [@{
+        NSParagraphStyleAttributeName: paragraphStyle,
+        NSFontAttributeName: font
+    } mutableCopy];
+    if (self.customFontAttributes[NSForegroundColorAttributeName] != nil) {
```

Comment:
Do you need to do a null check? I mean, given that this is a dictionary, you'd still be doing a null check when reading, but do you need to do so when writing?

Reply:
Are you sure?

Reply:
I take it back :D